src/lurk/eval.rs

```diff
@@ -2086,6 +2101,112 @@ pub fn eval_letrec<F: AbstractField>() -> FuncE<F> {
     )
 }
 
+pub fn collect_mutual_env<F: AbstractField>() -> FuncE<F> {
+    func!(
+        fn collect_mutual_env(binds_tag, binds): [2] {
```

Comment:
This function is recursive. Does it need to be tagged `partial`?

Reply:
I think this function only needs to be `partial` if we believe that a circular cons list (for the bindings) can be constructed inside the VM. Since I believe that is not the case (same case as a cyclic env), it's fine for this to not be partial. Same case as below